.github/workflows/build.yaml

```diff
@@ -20,6 +20,34 @@ on:
     - cron: '0 0 * * 1'
 
 jobs:
+  generate:
+    runs-on: ubuntu-latest
+    name: check go generate
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.x
+
+      - name: Install stringer
+        run: go install golang.org/x/tools/cmd/stringer@latest
+
+      - name: Run go generate
+        run: |
+          cd roc
+          go generate
+
+      - name: Check for git changes
+        run: |
+          git diff --quiet
+          if [[ $? -ne 0 ]]; then
+            echo "go generate might not have been run. Please run it and commit changes"
+            exit 1
+          fi
```

Comment:
Seems this check does not work and needs to be adjusted.

```
$ gh co 98
...
Switched to branch 'stringer_work'

$ make generate 
cd roc && go generate

$ git st
On branch stringer_work
Untracked files:
  (use "git add <file>..." to include in what will be committed)
	roc/channelset_string.go
	roc/clocksource_string.go
	roc/fecencoding_string.go
	roc/frameencoding_string.go
	roc/interface_string.go
	roc/loglevel_string.go
	roc/packetencoding_string.go
	roc/protocol_string.go
	roc/resamplerbackend_string.go
	roc/resamplerprofile_string.go

$ git diff --quiet ; echo $?
0
```

Reply:
Yeah i think using `git status --porcelain` would be better

Reply:
Yes, and likely we should analyze its output instead of checking exit code

Reply:
uh by analyzing output do you mean setting a flag when checking and adding an extra step in the job with if condition

Reply:
I thin k we just need a shell command that looks at git status output and fails if it detects modified or untracked files.

Reply:
Your current code looks good (if it works; I didn't test it)

```
if [[ $(git status --porcelain | grep -E "[MARCD].*\.go$") ]]; then
    echo "Run go generate and commit generated files"
    exit 1
  fi
  shell: /usr/bin/bash -e {0}
```